kreuzberg/_api/main.py

```diff
@@ -65,16 +68,118 @@ def exception_handler(request: Request[Any, Any, Any], exception: KreuzbergError
     )
 
 
+def _convert_value_type(current_value: Any, new_value: Any) -> Any:
+    if isinstance(current_value, bool):
+        if isinstance(new_value, str):
+            return str(new_value).lower() in ("true", "1", "yes", "on")
+        return bool(new_value)
+    if isinstance(current_value, int) and not isinstance(new_value, bool):
+        return int(new_value) if new_value is not None else current_value
+    if isinstance(current_value, float):
+        return float(new_value) if new_value is not None else current_value
+    return new_value
+
+
+def _config_to_dict(config: ExtractionConfig) -> dict[str, Any]:
```

Comment:
ExtractionConfig and all config types have `as_dict` method (should be defined via protocol or ABC) - use it instead of doing this manipulation.